### PR TITLE
Use build-std to decrease binary size

### PR DESCRIPTION
**Pull request by Gelbpunkt**
This is pretty much just a port of the gateway-queue changes to http-proxy.

Before:
```
localhost/old-http-proxy  latest      56980e1e6f65  7 minutes ago   4.9 MB
localhost/old-http-proxy  arm         1bdbfd419464  51 seconds ago  3.82 MB
```

After:
```
localhost/http-proxy      latest      b0c05aab90c1  8 minutes ago   4.7 MB
localhost/http-proxy      arm         d83412e1d209  4 seconds ago   3.66 MB
```

Cross compilation still works, tested locally.

### CODE REVIEW

**Review by AEnterprise — DISMISSED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -12,20 +12,22 @@ ARG FEATURES
...
         ln -s "/$MUSL_TARGET-cross/bin/$MUSL_TARGET-gcc" "/usr/bin/$MUSL_TARGET-gcc" && \
         ln -s "/$MUSL_TARGET-cross/bin/$MUSL_TARGET-ld" "/usr/bin/$MUSL_TARGET-ld" && \
         ln -s "/$MUSL_TARGET-cross/bin/$MUSL_TARGET-strip" "/usr/bin/actual-strip" && \
-        mkdir -p /app/.cargo && \
-        echo -e "[target.$RUST_TARGET]\nlinker = \"$MUSL_TARGET-gcc\"\nrustflags = \"-Zgcc-ld=lld\"" > /app/.cargo/config; \
+        GCC_VERSION=$($MUSL_TARGET-gcc --version | grep gcc | awk '{print $3}') && \
+        echo -e "[build]\nrustflags = [\"-L\", \"native=/$MUSL_TARGET-cross/$MUSL_TARGET/lib\", \"-L\", \"native=/$MUSL_TARGET-cross/lib/gcc/$MUSL_TARGET/$GCC_VERSION/\", \"-l\", \"static=gcc\", \"-Z\", \"gcc-ld=lld\"]\n[target.$RUST_TARGET]\nlinker = \"$MUSL_TARGET-gcc\"\n[unstable]\nbuild-std = [\"std\", \"panic_abort\"]\n" > /app/.cargo/config; \
     else \
         echo "skipping toolchain as we are native" && \
+        echo -e "[build]\nrustflags = [\"-L\", \"native=/usr/lib\"]\n[unstable]\nbuild-std = [\"std\", \"panic_abort\"]\n" > /app/.cargo/config && \
```
Comment (Erk-) [inline Dockerfile]: Would it be possible to insert some newlines here to make it more readable?

**Review by Erk- — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (Gelbpunkt) [reply Dockerfile]: done

**Review by AEnterprise — APPROVED** [review]

**Review by Erk- — APPROVED** [review]

**Comment by Erk-** [comment]
I assume it is a port of https://github.com/twilight-rs/gateway-queue/pull/8 right?

**Comment by Gelbpunkt** [comment]
> I assume it is a port of [twilight-rs/gateway-queue#8](https://github.com/twilight-rs/gateway-queue/pull/8) right?

Yes, it includes the build-std fixes discovered there.

**Pull request merged**